### ISSUE TITLE
unecessary-else false complaint where else cannot be emitted without changing behaviour

### ISSUE THREAD

**Issue by felixfbecker** [palantir/tslint#4623]
### Bug Report

- __TSLint version__: 5.15.0
- __TypeScript version__: 3.4.1
- __Running TSLint via__: (pick one) CLI

#### TypeScript code being linted

```ts
if (inputField === 'patterns') {
                        // Patterns should be added to the query as-is.
                        fieldsQueryParts.push(inputValue)
                    } else if (inputField === 'exactMatch') {
                        // Exact matches don't have a literal field operator (e.g. exactMatch:) in the query.
                        fieldsQueryParts.push(formatFieldForQuery('', inputValue, true))
                    } else if (inputField === 'type' && inputValue === 'code') {
                        // code searches don't need to be specified.
                        continue
                    } else {
                        fieldsQueryParts.push(formatFieldForQuery(inputField, inputValue))
                    }
```

with `tslint.json` configuration:

```json
"unecessary-else": true
```

#### Actual behavior

`The preceding `if` block ends with a `continue` statement. This `else` is unnecessary. (unnecessary-else) tslint(1)`

#### Expected behavior

This should not complain. Removing the `else` would cause different behaviour, because the previous `if` branches would fall through after their execution and also execute the code that's in the `else` branch.

**Comment by adidahiya** [comment]
just released 5.15.0, try that out, it has some bugfixes for this rule

**Comment by felixfbecker** [comment]
Yeah this is on the latest version

**Comment by felixfbecker** [comment]
Sorry, I think I got an old version from cache
